src/bosh.js

```diff
@@ -78,6 +78,7 @@ Strophe.Request.prototype = {
      *
      *  Throws:
      *    "parsererror" - A parser error occured.
+     *    "badformat" - The entity has sent XML that cannot be processed.
      *
```

Comment:
Please change `badformat` to `wrongformat`. Text isn't inherently bad ;)

Also, the exception description could rather be something like: `The entity responded in plain text whereas XML is required`.